pkg/reconciler/route/reconcile_resources.go

```diff
@@ -173,30 +215,86 @@ func (c *Reconciler) reconcilePlaceholderServices(ctx context.Context, route *v1
 	return services, nil
 }
 
-func (c *Reconciler) updatePlaceholderServices(ctx context.Context, route *v1.Route, services []*corev1.Service, ingress *netv1alpha1.Ingress) error {
+func (c *Reconciler) updatePlaceholderServices(ctx context.Context, route *v1.Route, pairs []resources.ServicePair, ingress *netv1alpha1.Ingress) error {
 	logger := logging.FromContext(ctx)
 	ns := route.Namespace
 
 	eg, egCtx := errgroup.WithContext(ctx)
-	for _, service := range services {
-		service := service
+	for _, pair := range pairs {
+		from := pair
```

Comment:
General preference that shadowing can be a bit subtle when reading it later

Reply:
I would argue this is one case where shadowing is better, since it's the fairly canonical way of working around go's reuse of the loop var when spawning goroutines (see e.g. [1](https://github.com/knative/serving/blob/9ac4acb72afd2b48431a6549da536a3de6c2f1bf/vendor/knative.dev/pkg/leaderelection/chaosduck/main.go#L156), [2](https://github.com/knative/serving/blob/fdfe3f236eee5565ee51b8d9a0525d0ffd26e6c7/test/e2e/destroypod_test.go#L298), [3](https://github.com/knative/serving/blob/d5d489c0babd5c9ae9a37d707c06df92d70711f5/pkg/reconciler/route/reconcile_resources.go#L182), [4](https://github.com/knative/serving/blob/3b198ecba4eeea41ca4dd9f553fcf89867c1d253/test/conformance/runtime/container_test.go#L104) in our code base - every example I could find of a loop var above an eg.Go or a t.Parallel, on a quick search, in our code base, shadows, fwiw)

Reply:
Probably a left over when refactoring - I'm good with shadowing with the same name